### ISSUE TITLE
Show upload progress bar while uploading

### ISSUE THREAD

**Issue by DexrnZacAttack** [AlekEagle/cumulonimbus-frontend#26]
I feel this would be a very nice thing since currently you just sit on a loading screen with no knowledge of how much the file has been uploaded

**Comment by AlekEagle** [comment]
I would love to implement this, however the modern fetch API in browsers does not have a way to measure progress for uploading files, I would have to switch to the old XMLHttpRequest API.

**Comment by AlekEagle** [comment]
I have a plan in the works to replace fetch for XMLHttpRequest on the frontend and add a progress bar. 

The fetch function does have an experimental API to use read streams so you can monitor how much of the stream has been consumed by the upload. Unfortunately, that API is only available in Chromium based browsers, leaving everyone else without any information on upload progress.

**Comment by DexrnZacAttack** [comment]
> I have a plan in the works to replace fetch for XMLHttpRequest on the frontend and add a progress bar. 
> 
> The fetch function does have an experimental API to use read streams so you can monitor how much of the stream has been consumed by the upload. Unfortunately, that API is only available in Chromium based browsers, leaving everyone else without any information on upload progress.

What about using fetch for the browsers that DO support it and XMLHttpRequest for the ones that don't... afaik I think you should be able to determine browser support for something like that right?

**Comment by AlekEagle** [comment]
yes, determining browser support would be simple, my only issue with that is I ~~am lazy~~ would like to write as little code as possible to keep the end bundle size small.

One of the main reasons I removed the legacy polyfill from the bundle was because it tripled the bundle size and doubled build times, so less code = more betterer load times.

I know the addition of both the XMLHttpRequest and fetch wouldn't be as detrimental as legacy polyfills, but streams in fetch have been in Chromium based browsers since around version 80, if widespread adoption was on its way we would have seen it by now in the other browsers, and since XMLHttpRequest isn't deprecated yet, I don't see that changing soon.

TL;DR: I'm lazy

**Comment by AlekEagle** [comment]
This will be addressed by ClickUp issue #86dr3e7mr

**Comment by AlekEagle** [comment]
It turns out that even if I use an XMLHttpRequest inside of the upload view itself, the fact that it is then routed out to the API via the service worker (the service worker converts all requests to fetch API) negates the fact that I used an XHR request in the actual app. I think what I will do is implement the Chromium only method, and hope that it gets adopted by other browsers eventually.

**Comment by AlekEagle** [comment]
Added in adeb2681be6e737810b7f63d137cae98abed6552
